BaseballPlayerImpl.cpp

```diff
@@ -9,7 +8,7 @@ BaseballPlayer::BaseballPlayerImpl::BaseballPlayerImpl(std::string a_name) :
 BaseballPlayer::BaseballPlayerImpl::~BaseballPlayerImpl() {}
 
 bool BaseballPlayer::BaseballPlayerImpl::has_baseball() const {
-    return this->baseball;
+    return baseball;
```

Comment:
The data member really should have a boolean-sounding name, like `has_baseball_`